_includes/css/activation.scss.liquid

```diff
@@ -37,89 +34,108 @@
-    {%- assign activation_first_level_reversed = activation_page.child_nav_order -%}
-    {%- break -%}
-  {%- endif -%}
-{%- endfor -%}
+{%- assign nav_list_link_prefix =
+      site_nav | split: nav_list_link | first | append: nav_list_link -%}
 
-{%- if activation_first_level_index == nil -%}
+{%- assign nav_splits =
+      nav_list_link_prefix | split: nav_list_item | pop -%}
 
-{{ activation_no_nav_link }}
+{%- comment -%}
+  The string before the first nav-list-item must contain a ul.nav-list tag.
+  It could start a list of ordinary pages or external links, or a collection.
+  
+  The string after a nav-list-item is non-empty, and may contain:
+  - a ul.nav-list tag (when the item is the first in a list) and/or
+    one or more closing ul tags (when the item is preceded by a list)
+  - a div.nav-category tag and a ul.nav-list tag (when the item is the first in
+    a foldable collection)
+  - neither of the above (when it is not the first item in a list).
+  
+  The extra list level due to foldable collections results in a superflous zero
+  item in nav_levels, which needs to be discarded.
+  
+  An empty collection does not contain nav-list-items. The nav_split containing
+  the navigation for N empty collections has corresponding occurrences of
+  nav_list with matching "</ul>"s, requiring addition of N to nav_levels[0].
+{%- endcomment %}
 
-{%- else -%}
+{%- assign nav_levels = "" | split: "" | push: 0 -%}
 
-{%- if page.grand_parent -%}
-  {%- assign activation_title = page.parent -%}
-  {%- assign activation_second_level = activation_pages
-        | where_exp: "item", "item.grand_parent == nil"
-        | where_exp: "item", "item.parent == page.grand_parent" -%}
-{%- elsif page.parent -%}
-  {%- assign activation_title = page.title -%}
-  {%- assign activation_second_level = activation_pages
-        | where_exp: "item", "item.grand_parent == nil"
-        | where_exp: "item", "item.parent == page.parent" -%}
-{%- endif -%}
-{%- if page.parent -%}
-  {%- include sorted_pages.html pages = activation_second_level -%}
-  {%- for activation_page in sorted_pages -%}
-    {%- if activation_page.title == activation_title -%}
-      {%- assign activation_second_level_index = forloop.index -%}
-      {%- assign activation_second_level_reversed = activation_page.child_nav_order -%}
-      {%- if activation_first_level_reversed -%}
-        {%- assign activation_second_level_index = sorted_pages | size | plus: 1 | minus: activation_second_level_index -%}
-      {%- endif -%}
-      {%- break -%}
+{%- for nav_split in nav_splits -%}
+
+    {%- assign nav_listless = nav_split | remove: nav_list -%}
+    {%- assign nav_lists = nav_split.size | minus: nav_listless.size | divided_by: nav_list.size -%}
+    {%- assign nav_endless = nav_split | remove: "</ul>" -%}
+    {%- assign nav_ends = nav_split.size | minus: nav_endless.size | divided_by: 5 -%}
+    
+    {%- if nav_lists >= 2 or nav_lists >= 1 and forloop.first or forloop.last -%}
```

Comment:
Suggestion: could you add a comment here that explains what the conditions are checking for?

Reply:
I'll try – but the case analysis that underlies the current code is unfortunately quite complicated…